From 26ecdfe466b5556ba6e90b92ec1b72b26eb5a539 Mon Sep 17 00:00:00 2001
From: ejaaskel <esa.jaaskela@suomi24.fi>
Date: Thu, 30 Oct 2025 13:08:30 +0200
Subject: [PATCH] Do not drop into CLI if not unlocked

By default, the system drops into the command line interface if the
autoboot fails. Do not do this if the device is not in the unlocked
state.

Upstream-Status: Pending
---
 common/main.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/common/main.c b/common/main.c
index 7c70de2e59a..ef1733af888 100644
--- a/common/main.c
+++ b/common/main.c
@@ -67,6 +67,12 @@ void main_loop(void)
 
 	autoboot_command(s);
 
-	cli_loop();
+	if (IS_ENABLED(CONFIG_UNLOCKING)) {
+		if (device_is_unlocked()) {
+			cli_loop();
+		}
+	} else {
+		cli_loop();
+	}
 	panic("No CLI available");
 }
-- 
2.34.1

